### DISCUSSION TITLE
How to register 'tw' and 'setup' globally ??

### DISCUSSION THREAD

**Discussion by Cappuuuu** [tw-in-js/twind#82] [Q&A] [answered]
Excuse me , How to register 'tw' and 'setup' globally ? 
I use Nuxt.js in my project .
Then i want to directly use `${tw}`  in HTML template  in `.vue`  without  ` import { tw, setup } from 'twind'`  in every  `.vue`

**Comment by lukejacksonn** [comment]
Hey 👋 one important things to note here is that `tw` works as a singleton meaning that you only ever have to call `setup` once in your app, just make sure it is before the first call to `tw`. Once it is setup then you can just `import { tw } from 'twind'` in your .vue files and the `tw` will use the config you specified during setup.

If you don't want to even have to import `tw` everywhere then you have two options I think:

1. Bind the `tw` function to window like `window.tw = tw`
2. Use the [twind/shim](https://github.com/tw-in-js/twind/blob/main/docs/installation.md#twindshim) which means you never even have to write `tw` you can just use `class="bg-red-500"`

**Reply by Cappuuuu** [reply]
Thanks ! I would try 😃

**Comment by tylerforesthauser** [comment, marked as answer]
Hey hey! I'd like to just jump in here too since I am also a Nuxt user who's aspiring to become a Twind user. I've thrown together a quick demo of one way you could go about using Twind anywhere in your Nuxt app.

https://codesandbox.io/s/nuxtjs-twind-s90bw

The approach is to create a `tw` instance in a plugin (`~/plugins/twind.js`) and it's here you can include your config. You then inject that instance into your Nuxt app for use virtually anywhere. I've elected to simply use a computed property so I could style directly in my template.

```html
<template>
  <h1 :class="tw('text-4xl')">
    Hello
  </h1>
</template>
```

If it's of any use, please let me know.

**Reply by sastan** [reply]
@tylerforesthauser Could you create an example for the docs?

**Reply by tylerforesthauser** [reply]
I'm still currently sorting out the whole server-side thing... But yes, I would love to put one together. I just figure I'd be best to include SSR support since that's often the whole point of Nuxt. I will try to do one for Vue as well...

**Reply by tylerforesthauser** [reply]
Alrighty, so the issue I've run into is that `node-html-parser` strips comments by default and this pisses Nuxt off because it causes hydration failure. Would it be feasible to allow user configuration of that [particular option](https://github.com/taoqf/node-html-parser#parsedata-options) in `twind/shim`? If I'm not mistaken, it would be required for both Vue SSR and Nuxt Universal apps.

**Reply by sastan** [reply]
Sure. Could you please create an issue. If you want you can try to adjust src/shim/server/index.ts and create an PR.

**Reply by Cappuuuu** [reply]
Thanks , it's works ! 😃
